### PR TITLE
refactor(db): unify table macros

### PR DESCRIPTION
**Pull request by DaniPopes**
Allows for easier modification of `Table` trait implementations and the `Tables` enum.

Based on https://github.com/paradigmxyz/reth/pull/6551

### CODE REVIEW

File: crates/storage/db/src/tables/mod.rs [hunk 1]
```diff
@@ -97,55 +94,96 @@ pub trait TableViewer<R> {
...
-        #[derive(Debug, PartialEq, Copy, Clone)]
-        /// Default tables that should be present inside database.
-        pub enum Tables {
+    (@bool) => { false };
+    (@bool $($t:tt)+) => { true };
+
+    (@view $name:ident $v:ident) => { $v.view::<$name>() };
+    (@view $name:ident $v:ident $_subkey:ty) => { $v.view_dupsort::<$name>() };
+
+    ($( $(#[$attr:meta])* table $name:ident<Key = $key:ty, Value = $value:ty $(, SubKey = $subkey:ty)? $(,)?>; )*) => {
```
Comment (shekhirin) [inline crates/storage/db/src/tables/mod.rs:112]: do we need this `table` marker token here? I see it's repeated for every table below

Comment (DaniPopes) [reply crates/storage/db/src/tables/mod.rs:112]: Looks nicer :D

**Comment by DaniPopes** [comment]
The assertion ` assert_eq!(table as usize, table_index);` is failing since the declaration order is different now because before all dupsort tables were declared at the bottom.

I'm assuming variant order does not matter since it never implemented Ord, so I will just remove this check.

**Review by mattsse — APPROVED** [review]
nice, lgtm

**Pull request merged**